lib/purescript-cst/src/Language/PureScript/Types.hs

```diff
@@ -525,8 +525,10 @@ unknowns = everythingOnTypes (<>) go where
   go (TUnknown _ u) = IS.singleton u
   go _ = mempty
 
+-- | Check if a type contains unknowns in a position that is relevant to
+-- constraint solving. (Kinds are not.)
 containsUnknowns :: Type a -> Bool
-containsUnknowns = everythingOnTypes (||) go where
+containsUnknowns = everythingOnTypes (||) go . eraseKindApps where
```

Comment:
My take is that non-ambiguous kind polymorphism on classes are equivalent to adding extra arguments to the class with fundeps pointing from the type to the kind. That is:

```purescript
class Wat :: forall k. k -> Constraint
class Wat ok
```
Would be kind of like saying:
```purescript
class Wat k ok | ok -> k
```
for the purposes of dispatch.

In the case of ambiguous kind variables, it's not really any different from a normal parameter, it would just be provided through kind applications.

Reply:
The kind-of-like-fundeps are the thing I'm asking about here; so if `ok` is known but `k` is unknown, and no instance of `Wat ok` is found, it's not for lack of type annotations, because `k` is determined by `ok`. And from that, I conclude that it's correct to strip out kinds from types before checking those types for unknowns when deciding whether to show the ‘consider adding type annotations’ hint.

Reply:
So... is this a safe change to make?

Reply:
I stand by my above comments. More to the point of safety, though, this is a change to a function that is only ever used to determine whether or not to display a particular hint in an error message; the harm of getting this wrong is minimal.